tornado/httputil.py

```diff
@@ -230,13 +239,127 @@ def parse(cls, headers):
            mix of `KeyError`, and `ValueError`.
 
         """
-        h = cls()
+        h = HTTPHeaders()
         for line in headers.splitlines():
             if line:
                 h.parse_line(line)
         return h
 
-    # MutableMapping abstract method implementations.
+    def pop(self, key, default=__marker):
```

Comment:
Shouldn't this method also update _last_key ?

Reply:
Why?

`_last_key` is only used for parsing headers, which happens with `add` exclusively, so I don't see a need

Reply:
The sequence I had in mind was something like:

- add (key) => self._last_key is set to key
- pop (key) => self._last_key not updated
- parse_line (line) => KeyError is raised because they key is no longer present

I'm not sure if that usage is valid (although I think it should), but it seems to me like it may present problems.

Reply:
The original implementation with MutableDict also had that issue it would seem, so I don't think it's a problem